horace_core/sqw/axes_blocks/@AxesBlockBase/build_from_input_binning.m

```diff
@@ -131,6 +131,11 @@
                 range = [range(1),range(3)];
             end
         end
+    otherwise
+        error('HORACE:AxesBlockBase:invalid_argument',[ ...
+            '*** Binning arguments can be provided in four possible forms:\n' ...
+            '*** []; step; [integration_min,integration_max]; [bin_min,step,bin_max]\n' ...
+            '*** You have provided data with %d elements namely: %s'],numel(bin_req),disp2str(bin_req));
```

Comment:
this is very condensed. You say there are 4 possible forms but you run all of them into 1 line with various defaults. Lay out the 4 options separately.
Then line 138 should just give the actual input and say it does not meet any of these 4 forms. By pointing to a specific structure of the wrong input, you imply that some diagnosis has taken place. Let the user figure out what they should do to meet the requirements.